packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -737,17 +737,29 @@ class FlutterPlugin implements Plugin<Project> {
         // compile/target/min sdk values.
         pluginProject.extensions.create("flutter", FlutterExtension)
 
-        // Add plugin dependency to the app project.
-        project.android.buildTypes.each { buildType ->
-            String flutterBuildMode = buildModeFor(buildType)
-            if (flutterBuildMode != "release" || !pluginObject.dev_dependency) {
-                // Only add dependency on dev dependencies in non-release builds.
-                project.dependencies {
-                    api(pluginProject)
+        project.android.buildTypes {
+            // Add profile build type since it may not already exist.
+            profile {
+                initWith(debug)
+                if (it.hasProperty("matchingFallbacks")) {
+                    matchingFallbacks = ["debug", "release"]
```

Comment:
I'm not 100% confident on this. TLDR is that the profile mode has to exist in order for us to specify `profileApi`, but typically it's defined after this method is called. I added it redundantly here to make sure it exists before. I left the other one because I assume it needs it for the case where there are no plugins, but I'll double check.

Reply:
If you intend to double check before submitting, ACK, otherwise let's add a TODO/GH issue as well before submitting.

Reply:
oh ack! I checked. I think based on how other things are written in the plugin, this should be fine.

Reply:
Had an idea to slightly clean this up so gonna do that before this lands